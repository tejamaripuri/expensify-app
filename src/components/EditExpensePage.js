import React from 'react';

const EditExpensePage = (props) => (
    <div>
        Editing the {props.match.params.id}.
    </div>
);

export default EditExpensePage;