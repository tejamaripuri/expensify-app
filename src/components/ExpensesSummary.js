import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formatedExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totaling <span>{formatedExpenseTotal}</span></h1>
                <div className="page-header__action">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStatetoProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }; 
};

export default connect(mapStatetoProps)(ExpensesSummary);