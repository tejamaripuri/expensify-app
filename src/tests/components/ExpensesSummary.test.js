import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { wrap } from 'module';

test('should render for multiple expenses correctly.', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={12345} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render for single expenses correctly.', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1500} />);
    expect(wrapper).toMatchSnapshot();
});