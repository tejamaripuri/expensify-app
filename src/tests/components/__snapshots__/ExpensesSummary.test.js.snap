// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render for multiple expenses correctly. 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totaling 
    $123.45
  </h1>
</div>
`;

exports[`should render for single expenses correctly. 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $15.00
  </h1>
</div>
`;
